mlflow/projects/yarn.py

```diff
@@ -0,0 +1,288 @@
+        {
+            'SKEIN_CONFIG': './.skein',
+            'PEX_ROOT': "./.pex",
+            "PYTHONPATH": ".",
+            "GIT_PYTHON_REFRESH": "quiet"
+        }
+    )
+
+    dict_files_to_upload = {os.path.basename(path): os.path.abspath(path)
+                            for path in additional_files}
+
+    python_bin = f"./{os.path.basename(pex_env)}" if pex_env.endswith(
+        '.pex') else f"./{os.path.basename(pex_env)}/bin/python"
+
+    launch_options = "-m" if not module_name.endswith(".py") else ""
+    launch_args = args if args else ""
+    _logger.info("""
+                    set -x
+                    ls -al .
+                    env
+                    export HADOOP_CONF_DIR=%s
+                    %s %s %s %s
+                """, hadoop_conf_dir, python_bin, launch_options, module_name, launch_args)
+
+    _logger.info("ENV DICT = %s", env)
+    _logger.info("ADDITIONAL FILES = %s", dict_files_to_upload)
+
+    service = skein.Service(
+        resources=skein.model.Resources(memory, num_cores),
+        instances=num_containers,
+        files=dict_files_to_upload,
+        env=env,
+        script="""
+                    set -x
+                    ls -al .
+                    env
+                    export HADOOP_CONF_DIR=%s
+                    %s %s %s %s
+                """ % (hadoop_conf_dir, python_bin, launch_options, module_name, launch_args)
+    )
+
+    _logger.info("FILE_SYSTEMS = %s", hadoop_file_systems)
+    spec = skein.ApplicationSpec(
+        name=name,
+        file_systems=hadoop_file_systems,
+        services={name: service},
+        acls=skein.model.ACLs(
+            enable=True,
+            ui_users=['*'],
+            view_users=['*']
+        )
+    )
+
+    if user:
+        spec.user = user
+
+    if queue:
+        spec.queue = queue
+
+    if node_label:
+        service.node_label = node_label
+
+    print("SKEIN APP SPEC: %s" % spec.to_json())
+
+    return skein_client.submit(spec)
+
+
+def _merge_env_lists(env_params, env_yarn_cfg):
+    env = dict(value.split('=') for value in env_params)
+    env.update(dict(value.split('=') for value in env_yarn_cfg))
+    return env
+
+
+def _get_key_from_params(params, key, remove_key=True):
+    if key not in params:
+        return []
+
+    values = params[key].split(',')
+    if remove_key:
+        del params[key]
+
+    return values
+
+
+def _parse_yarn_config(backend_config, extra_params=None):
+    """
+    Parses configuration for yarn backend and returns a dictionary
+    with all needed values. In case values are not found in ``backend_config``
+    dict passed, it is filled with the default values.
+    """
+
+    extra_params = extra_params or {}
+    if not backend_config:
+        raise ExecutionException("Backend_config file not found.")
+    yarn_config = backend_config.copy()
+
+    for cfg_key in [YARN_NUM_CORES, YARN_MEMORY, YARN_QUEUE,
+                    YARN_HADOOP_FILESYSTEMS, YARN_HADOOD_CONF_DIR,
```

Comment:
Typo here: YARN_HADOOD_CONF_DIR

Reply:
fixed in c8f11f0128494c47abbb0de03b5305c06b568708